chronos/asyncloop.nim

```diff
@@ -228,17 +229,19 @@ func getAsyncTimestamp*(a: Duration): auto {.inline.} =
     result += min(1, cast[int32](mid))
 
 template processTimersGetTimeout(loop, timeout: untyped) =
-  var count = len(loop.timers)
-  if count > 0:
-    var lastFinish = curTime
-    while count > 0:
-      lastFinish = loop.timers[0].finishAt
-      if curTime < lastFinish:
-        break
-      loop.callbacks.addLast(loop.timers.pop().function)
-      dec(count)
-    if count > 0:
-      timeout = (lastFinish - curTime).getAsyncTimestamp()
+  var lastFinish = curTime
+  while loop.timers.len > 0:
+    if loop.timers[0].deleted:
+      discard loop.timers.pop()
+      continue
+
+    lastFinish = loop.timers[0].finishAt
+    if curTime < lastFinish:
+      break
+
+    loop.callbacks.addLast(loop.timers.pop().function)
+  if loop.timers.len > 0:
```

Comment:
With such approach logic was changed, so `loop.timers.len` will be always `0` at this point, but `count` can be not `0`.

Reply:
Not exactly sure what you're saying, there is a break on line 240 and only if the condition of the break never hits, will `len` reach `0`.

Reply:
Please take a look on old code, `count` is set to number of items in timers queue, but its possible that inside of `while loop.timers.len > 0` you will clean all the timers queue and so `if loop.timers.len > 0:` will never be executed, while old code will be executed because it uses saved value `count` of queue length.

Reply:
My comment is wrong here, this place is fine.